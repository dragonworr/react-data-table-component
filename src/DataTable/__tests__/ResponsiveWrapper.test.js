import 'jest-styled-components';

import React from 'react';
import { shallow } from 'enzyme';

import ResponsiveWrapper from '../ResponsiveWrapper';

test('component <ResponsiveWrapper /> should render correctly', () => {
  const wrapper = shallow(<ResponsiveWrapper />);
  expect(wrapper).toMatchSnapshot();
});

test('component <ResponsiveWrapper responsive /> should render correctly ', () => {
  const wrapper = shallow(<ResponsiveWrapper responsive />);
  expect(wrapper).toMatchSnapshot();
});

test('component <ResponsiveWrapper overflowY /> should not apply overFlowY without an overflowYOffset or not responsive ', () => {
  const wrapper = shallow(<ResponsiveWrapper overflowY />);
  expect(wrapper).toMatchSnapshot();
});

test('component <ResponsiveWrapper responsive overflowY overflowYOffset="250px" /> should render correctly when overflowYOffset is provided ', () => {
  const wrapper = shallow(<ResponsiveWrapper responsive overflowY overflowYOffset="250px" />);
  expect(wrapper).toMatchSnapshot();
});
