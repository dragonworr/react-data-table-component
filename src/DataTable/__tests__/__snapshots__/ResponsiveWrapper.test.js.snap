// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <ResponsiveWrapper /> should render correctly 1`] = `
.c0 {
  position: relative;
}

<div
  className="c0"
/>
`;

exports[`component <ResponsiveWrapper overflowY /> should not apply overFlowY without an overflowYOffset or not responsive  1`] = `
.c0 {
  position: relative;
}

<div
  className="c0"
/>
`;

exports[`component <ResponsiveWrapper responsive /> should render correctly  1`] = `
.c0 {
  position: relative;
  overflow-x: auto;
}

<div
  className="c0"
/>
`;

exports[`component <ResponsiveWrapper responsive overflowY overflowYOffset="250px" /> should render correctly when overflowYOffset is provided  1`] = `
.c0 {
  position: relative;
  overflow-x: auto;
  padding-bottom: 250px;
  margin-bottom: -250px;
}

<div
  className="c0"
/>
`;
